src/main/ruby/truffleruby/core/truffle/file_operations.rb

```diff
@@ -78,5 +78,35 @@ def self.expand_path(path, dir, expand_tilde)
     def self.exist?(path)
       Truffle::POSIX.truffleposix_stat_mode(path) > 0
     end
+
+    #Helper to File.dirname, extracts the directory of a path (doesn't need to handle the empty string)
+    #This was the original File.dirname, but ruby 3.1 made File.dirname take an optional number of times to do its logic
+    #So the original method now becomes calling this in a loop, hence the renaming with "_once" appended
+    def self.dirname_once(path)
```

Comment:
TBH the method name is a bit confused. I would consider something more obvious like "dirname_remove_last_segment" or "...without_last_segment" etc

Reply:
Hmm, I think the name is just a nod to the fact that this where most of the logic for File.dirname is performed, and the visible File.dirname is just calling it in a loop, I agree it seems confusing at first glance. 

I think "dirname_without_last_segment" is redundant, since the "dirname" part already implies the "without_last_segment", right ? Ideally we want something that have 'dirname' and then some indicator of "oneness" to imply that it will do so only once, or maybe we should ditch "dirname" entirely and just rename it like "remove_last_segment_from_path" ?

Reply:
I like the term "path" here 👍 .

I don't have strong opinion on this subjective topic and I am OK with any name that makes it obvious what a method does without looking through its code.

Reply:
I would just name it `dirname` since it was `dirname(1)` does.
The fact it's under `Truffle::FileOperations` implies it's a helper method and doesn't do all the logic, and one can easily look at `File.dirname` to see how they use each other.